Sources/NIO/SelectorEpoll.swift

```diff
@@ -93,6 +93,35 @@ extension SelectorEventSet {
     }
 }
 
+@usableFromInline struct EPollUserData {
+    // We're backing this by a tuple to get standard C layout
+    @usableFromInline var registrationID: SelectorRegistrationID
+    @usableFromInline var fileDescriptor: CInt
+
+
+    @inlinable init(registrationID: SelectorRegistrationID, fileDescriptor: CInt) {
+        assert(MemoryLayout<UInt64>.size == MemoryLayout<EPollUserData>.size)
+        self.registrationID = registrationID
+        self.fileDescriptor = fileDescriptor
+    }
+
+    @inlinable init(rawValue: UInt64) {
+        let registrationID = UInt32(truncatingIfNeeded: (rawValue & (UInt64(UInt32.max) << 32)) >> 32)
```

Comment:
We don't need the mask here: shifting right by 32 is the same as masking off the low 32 bits and then shifting by 32.

Reply:
replaced by generic version, code is gone